src/datastore/pg-write-store.ts

```diff
@@ -552,6 +552,32 @@ export class PgWriteStore extends PgStore {
     return result.count;
   }
 
+  async insertStxEventBatch(sql: PgSqlClient, stxEvents: StxEventInsertValues[]) {
+    const values = stxEvents.map(s => {
+      const value: StxEventInsertValues = {
+        event_index: s.event_index,
+        tx_id: s.tx_id,
+        tx_index: s.tx_index,
+        block_height: s.block_height,
+        index_block_hash: s.index_block_hash,
+        parent_index_block_hash: s.parent_index_block_hash,
+        microblock_hash: s.microblock_hash,
+        microblock_sequence: s.microblock_sequence,
+        microblock_canonical: s.microblock_canonical,
+        canonical: s.canonical,
+        asset_event_type_id: s.asset_event_type_id,
+        sender: s.sender,
+        recipient: s.recipient,
+        amount: s.amount,
+        memo: s.memo ?? null,
+      };
+      return value;
+    });
+    await sql`
+      INSERT INTO stx_events ${sql(values)}
```

Comment:
Perhaps these inserts need to be done with cursors. I'm not sure how many rows you're going to be inserting at a time (here and in the other examples below), but if they're going to be a large number you should check this out: https://github.com/porsager/postgres#cursors

Keep in mind that the current way you're doing it actually creates a prepared statement query before running it (with placeholders such as `$1, $2, $3, ...`) and there's a max number of placeholders postgres supports at a time so you could get some crashes here.

Reply:
> I'm not sure how many rows you're going to be inserting at a time

It depends. It's controlled by the size of the batches.

> there's a max number of placeholders postgres supports at a time so you could get some crashes here.

Yep... if I am not wrong postgres supports 64K of parameters in a statement. I've faced this limit and it's being controlled by the batch size.